src/stream/salsa20/xsalsa20_test.c

```diff
@@ -0,0 +1,89 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+/* The implementation is based on:
+ * "Extending the Salsa20 nonce", https://cr.yp.to/snuffle/xsalsa-20081128.pdf
+ * "Salsa20 specification", http://cr.yp.to/snuffle/spec.pdf
+ * and salsa20-ref.c version 20051118
+ * Public domain from D. J. Bernstein
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_XSALSA20
+
+int _sha256(unsigned char *hash, const unsigned char *data, const int datalen) {
+   hash_state md;
+   sha256_init(&md);
```

Comment:
pretty paranoid, but this would fail to build if someone wants to build w/o sha256

I think it'd be better to use the `hash_memory()` function instead of this `_sha256()` ... but I just realized then again we'd have a dependency on the `LTC_HASH_HELPERS` ... btw. how's this handled in all the other tests? :D 

Simply put this function and the second test-case in a `#ifdef LTC_SHA256` please